crowbar_framework/app/models/provisioner_service.rb

```diff
@@ -209,25 +195,30 @@ def synchronize_repositories(platforms)
 
   def enable_all_repositories
     @logger.debug("Enabling all repositories.")
-    bag = load_repository_bag
     Crowbar::Repository.check_all_repos.each do |repo|
-      enable_repository(repo.platform, repo.arch, repo.id, bag)
+      enable_repository(repo.platform, repo.arch, repo.id)
     end
   end
 
   def disable_all_repositories
     @logger.debug("Disabling all repositories.")
-    bag = load_repository_bag
-    bag.keys.each do |platform|
-      next if platform == "id"
-      bag.delete(platform)
+    all_db = begin
+      Chef::DataBag.list
+    rescue Net::HTTPServerException
+      []
     end
-    bag.save
-  end
 
-  def enable_repository(platform, arch, repo, bag = nil)
-    bag = load_repository_bag if bag.nil?
+    all_db.keys.each do |db_name|
+      next unless db_name =~ /^repos-.*/
+      begin
+        chef_data_bag_destroy(db_name)
+      rescue Net::HTTPServerException
+       @logger.debug("Cannot disable repos for #{db_name}!")
```

Comment:
Style/IndentationWidth: Use 2 (not 1) spaces for indentation. (https://github.com/bbatsov/ruby-style-guide#spaces-indentation)